### PR TITLE
Fix #2358 by recognizing props wrapped in $ReadOnly<>

### LINKED ISSUE #2358: no-unused-prop-types doesn't report errors on readonly props
If you wrap the props in $ReadOnly<> then no-unused-prop-types doesn't validate.

```
type Props = $ReadOnly<{| 
    a: string,
 |}>;
```

### PR DESCRIPTION
**Pull request by lukeapage**
Fixes #2358.

### CODE REVIEW

**Comment by lukeapage** [comment]
Not sure why the build fails - I don’t see how it could be related to my changes. The same error happened on master a month ago:

https://travis-ci.org/yannickcr/eslint-plugin-react/builds/551059139

**Comment by jomasti** [comment]
There are two ESLint errors related to your changes that are causing that failure. Lines 651-652 are specifically the cause of the failure.

**Comment by lukeapage** [comment]
Thanks! I'm not used to a project with many warnings, I missed the fact there were 2 errors.

File: lib/util/propTypes.js [hunk 1]
```diff
@@ -545,6 +545,15 @@ module.exports = function propTypesInstructions(context, components, utils) {
       case 'IntersectionTypeAnnotation':
         ignorePropsValidation = declarePropTypesForIntersectionTypeAnnotation(propTypes, declaredPropTypes);
         break;
+      case 'GenericTypeAnnotation':
+        if (propTypes.id.name === '$ReadOnly') {
+          ignorePropsValidation = declarePropTypesForObjectTypeAnnotation(
+            propTypes.typeParameters.params[0],
+            declaredPropTypes);
```
Comment (ljharb) [inline lib/util/propTypes.js]:
```suggestion
            declaredPropTypes
          );
```

**Comment by lukeapage** [comment]
okay, done

**Pull request merged**